core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java

```diff
@@ -103,7 +102,7 @@ private void addDimensions(List<CarbonDimension> dimensions) {
         } else {
           noSortNoDictDimSpec.add(spec);
         }
-      } else if (dimension.isDirectDictionaryEncoding()) {
+      } else if (dimension.getDataType() == DataTypes.DATE) {
         spec = new DimensionSpec(ColumnType.DIRECT_DICTIONARY, dimension, dictActualPosition++);
```

Comment:
ColumnType.DIRECT_DICTIONARY, we are still keeping ? we can refactor this also

Reply:
I will refactor ColumnType in another PR

Reply:
This is not possible in this PR since DATE is still a direct dictionary, need to do in another PR

Reply:
Please add it to PR description.